cabal-install/src/Distribution/Client/Init/Format.hs

```diff
@@ -87,21 +91,23 @@ fieldD fieldName fieldContents fieldComments includeField opts
         empty
 
     fieldSWithContents cs =
-      PrettyField (withComments (map ("-- " ++) cs)) fieldName fieldContents
+      PrettyField (withComments . isCabalVersion fieldName $ map ("-- " ++) cs) fieldName fieldContents
 
 
 -- | A field annotation instructing the pretty printer to comment out the field
 --   and any contents, with no comments.
-commentedOutWithComments :: [String] -> FieldAnnotation
-commentedOutWithComments = FieldAnnotation True . map ("-- " ++)
+commentedOutWithComments :: CommentPosition -> FieldAnnotation
+commentedOutWithComments (CommentBefore cs) = FieldAnnotation True . CommentBefore $ map ("-- " ++) cs
+commentedOutWithComments (CommentAfter  cs) = FieldAnnotation True . CommentAfter  $ map ("-- " ++) cs
+commentedOutWithComments NoComment = FieldAnnotation True NoComment
```

Comment:
I'm not very confident on this solution, it looks too ad-hoc-y. Maybe I could get rid of the `NoComment` value altogether?

Reply:
i'd actually argue this is a fairly clean solution